glpi-project/glpi
# Modularize and test dashboards JS

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Tests currently covers just under 90% of statements. Old, untested JS file was left in place and is still used by GLPI core to ensure BC. It now contains only a dynamic import statement which allows loading the ESM version of the dashboard JS from the non-module script.

Fixes:
- Potential warning when saving permissions when it tries splitting the value of the Is Private form input by '-' as if it were a sharing target like 'users_id-2'.
- Issue where a configured suffix is never actually appended to a number card in `animateNumbers` because the `+ suffix` part was outside of the `text()` function. I'm not sure anything uses suffixes currently.

Improvements to Jest testing:
- Support mocking a failure in the AJAX mocking system
- Support for `$.get` and `$.post` in the AJAX mocking system
- Add some output when no AJAX mock request is matched to help debug the issue
- Add coverage options to Jest config (disabled by default)
- Moved test directory option from NPM script to Jest config
